### PR TITLE
Add MockOnlyRule to allow skipping specific tests based on downloader

### PR DESCRIPTION
**Pull request by XiangRongLin**
- [x] I carefully read the [contribution guidelines](https://github.com/TeamNewPipe/NewPipe/blob/HEAD/.github/CONTRIBUTING.md) and agree to them.

### -
- addresses point 7 of #481
- from https://github.com/junit-team/junit4/issues/116

### showcase
- changes: https://github.com/XiangRongLin/NewPipeExtractor/commits/flaky_test_showcase
- Run that branch locally with `gradle check -Ddownloader=REAL` and gradle check -Ddownloader=MOCK`
  - Mock will fail because failing test is not skipped
  - REAL will succeed because failing test is skipped

### CODE REVIEW

**Comment by XiangRongLin** [comment]
After thinking about it a bit more i think adding a custom test runner is better.
- it can be specified globaly in gradle file (i think, could not find it immediatly). If not there is an additional overhead of specifiying it
- Can be easily added on method level, instead of class level

**Comment by XiangRongLin** [comment]
With rules it's also possible to skip tests on method level instead of class level. Other that that it's the same addtional overhead compared to the developer to add it. But it allows to use other test runner (Mockito).

I adjusted the showcase branch

File: extractor/src/test/java/org/schabi/newpipe/MockOnly.java [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+ *
+ * {@link MockOnlyRule}
+ */
+@Retention(RetentionPolicy.RUNTIME)
+@Target({ElementType.METHOD, ElementType.TYPE})
+@Inherited
+public @interface MockOnly {
+
+    /**
+     * Explanation why this test shold only be run with mocks and not against real websites
```
Comment (B0pol) [inline extractor/src/test/java/org/schabi/newpipe/MockOnly.java:22]:
```suggestion
     * Explanation why this test should only be run with mocks and not against real websites
```

Comment (XiangRongLin) [reply extractor/src/test/java/org/schabi/newpipe/MockOnly.java]: done

**Review by Stypox — APPROVED** [review]

**Comment by Stypox** [comment]
@XiangRongLin can this be merged? If yes, feel free to do it yourself 😉

**Pull request merged**